### DISCUSSION TITLE
Dataframe methods to trigger computations

### DISCUSSION THREAD

**Discussion by YarShev** [dask/dask#9395] [Q&A] [answered]
Hi guys,

I have a question regarding triggering computations in Dask. Let's consider the following example.

```python
import dask.dataframe as pd
dd = pd.read_csv("file")
pdf1 = dd.head() # 1st case
pdf2 = dd.compute() # 2st case
```
I understand when we call dd.compute() the reading of the whole dataset from a file is triggered. However, if we call dd.head(), whether the reading of the whole dataset from a file is triggered or only a subset of the data matching `n_partitions` parameter of the .head() method is read?

Thanks in advance!

**Comment by YarShev** [comment, marked as answer]
cc @martindurant, @jsignell

**Reply by jsignell** [reply]
Only a subset of data is read when you call `dask.DataFrame.head()` However if you are chaining together operations then the whole dataset might have to be read in order to get `.head()`. For instance imagine you are normalizing:

```python
normalized_ddf = ddf - ddf.mean()
normalized_ddf.head()
```

In order to return `.head()` Dask needs to first compute the mean of all the columns, and to do that, Dask needs to read all the data.

**Reply by YarShev** [reply]
Thank you @jsignell. That makes sense to me.
